### ISSUE TITLE
Vector buffers are not cleaned up

### ISSUE THREAD

**Issue by mr-karan** [vectordotdev/vector#10957]
### Vector Version

```
vector 0.19.0 (x86_64-unknown-linux-gnu da60b55 2021-12-28)
```

### Vector Configuration File

```toml
  batch.max_bytes = 1049000 # 1MB
  batch.timeout_secs = 10
  buffer.max_size = 10490000000 # 10GB
  buffer.type = "disk"
  buffer.when_full = "block"
```

### Expected Behavior

`vector` should release the batch pending in disk buffers. It doesn't until you restart.

### Actual Behavior

- I ran `vector` which processed logs (`input` is gzipped CSVs as the `file` source) and dumped to sink (`output` is `clickhouse`). After `vector` finished processing all the files, the disk size of `/var/lib/vector` was 42MB. (Unfortunately I don't have the contents of the data directory anymore). Since I have `remove_after_sec` configured, the files were deleted after `vector` read those files. No new data was written on these log files.

- I restarted `vector`. I could see the disk size of `/var/lib/vector` dropped from `42MB` to `4.3MB`. I also saw some logs being pushed to the `sink` (using `vector top`):

![vector_batch](https://user-images.githubusercontent.com/5689132/150519974-f7d203b1-68e4-4f0d-9124-a4338965255c.png)

This is really strange since the actual log file (input source) doesn't exists on the disk.

Which means that until I restarted, `vector` did have some pending logs in the buffer that it didn't send to the `sink`. I confirmed before stopping `vector` that it wasn't processing any events and it was idle (`vector top` showed 0 events in/out).

Due to this, I've to resort to a cron which restarts `vector` every `n` hours but ideally I'd like to avoid doing this. :)

**Comment by jszwedko** [comment]
Hi @mr-karan ! I think this is the same issue as https://github.com/vectordotdev/vector/issues/10806. The fix for it will be going out in 0.19.1 early next week.

**Comment by jszwedko** [comment]
I'll close this, but feel free to reopen if you think it is a different issue.
